### PR TITLE
use structured facts via hash $facts

### PR DESCRIPTION
**Pull request by Dan33l**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
   # this warning is never reached. If Puppet < 4 is used with tis module,
   # Puppet fail with error about data type like "Syntax error at 'String'; expected ')'"
-  if versioncmp($::puppetversion, '4.0.0') == -1 {
+  if versioncmp($facts['puppetversion'], '4.0.0') == -1 {
```
Comment (alexjfisher) [inline manifests/init.pp]: Time to remove this altogether? Version 2.0 dropping puppet 3 was out a long time ago now.

Comment (Dan33l) [reply manifests/init.pp]: Yes i planed to do this in an other PR. But if you agree we can do it now.

File: manifests/init.pp [hunk 2]
```diff
@@ -56,7 +56,7 @@
   if $manage_kernel {
     Class['virtualbox::install'] -> class { 'virtualbox::kernel': }
 
-    if $::osfamily == 'RedHat' {
+    if $facts['osfamily'] == 'RedHat' {
```
Comment (alexjfisher) [inline manifests/init.pp]: If you do a modulesync, I think this will then fail the `legacy_facts` check?

File: manifests/init.pp (hunk 1 above)
Comment (runejuhl) [reply manifests/init.pp]: I'd be fine with that :+1:

**Comment by Dan33l** [comment]
With structured facts tests are failing so, i switch unit tests to `rspec-puppet-facts`

**Pull request merged**